component/src/ibc/mod.rs

```diff
@@ -7,6 +7,10 @@ mod client;
 mod component;
 mod connection;
 pub(crate) mod event;
+
+#[allow(dead_code)]
+pub(crate) mod ibc_token;
+
```

Comment:
rather than having to do `allow(dead_code)`, if we did
```
mod ibc_token;
pub use ibc_token::IBCToken;
```
we'd make the type accessible and avoid the warning.